### DISCUSSION TITLE
SHACL2DTDL

### DISCUSSION THREAD

**Discussion by cbupp** [RealEstateCore/rec#209] [General]
@hammar - Reviewing the current pull requests, and different branches in the [SHACL2DTDL](https://github.com/RealEstateCore/SHACL2DTDL/branches) to support the different iterations. I'm wondering if it makes sense to maintain SHACL2DTDL as a separate repo? Or maybe at a minimum, to reference the specific version used to generate the DTDL as a git submodule and include it as part of the CI pipeline?

**Comment by hammar** [comment]
You're right on this, we should probably integrate this under /Tools/. I've been putting it off because SHACL2DTDL is a horrible horrible hack of a script that needs to be rewritten using the DTDL parser library -- which would probably cut it in half in terms of LOC, and speed it up significantly to boot. It's on my back log of things to look into when I have time. If I don't get to it, if you're feeling adventurous, feel free to take a stab at it and/or to look at the existing SHACL2DTDL code.
